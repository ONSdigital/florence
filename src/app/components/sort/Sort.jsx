import React from "react";
import clsx from "clsx";
import PropTypes from "prop-types";

const Sort = ({ active, name }) => {
    const { key, direction } = active;
    const isActive = (name, sortDirection) => {
        if (key === name && direction === sortDirection) return "active";
    };

    return (
        <span className="sort">
            <span data-testid="ASC" className={clsx("sort__icon-up", isActive(name, "ASC"))} />
            <span data-testid="DESC" className={clsx("sort__icon-down", isActive(name, "DESC"))} />
        </span>
    );
};

Sort.propTypes = {
    active: PropTypes.shape({
        key: PropTypes.string,
        direction: PropTypes.oneOf(["ASC", "DESC"]),
    }),
    name: PropTypes.string.isRequired,
};

export default Sort;
