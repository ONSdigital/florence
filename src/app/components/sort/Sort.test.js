import React from "react";
import renderer from "react-test-renderer";
import Sort from "./Sort";

describe("Sort", () => {
    // this component is dummy atm so checking snapshot is sufficient for regression checks only
    it("renders correctly with active class", () => {
        const props = {
            active: { key: "foo", direction: "ASC" },
            name: "foo",
        };
        const tree = renderer.create(<Sort {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders correctly without active class", () => {
        const props = {
            active: { key: "boo", direction: "DESC" },
            name: "foo",
        };
        const tree = renderer.create(<Sort {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
