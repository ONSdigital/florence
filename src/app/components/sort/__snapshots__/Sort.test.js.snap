// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort renders correctly with active class 1`] = `
<span
  className="sort"
>
  <span
    className="sort__icon-up active"
    data-testid="ASC"
  />
  <span
    className="sort__icon-down"
    data-testid="DESC"
  />
</span>
`;

exports[`Sort renders correctly without active class 1`] = `
<span
  className="sort"
>
  <span
    className="sort__icon-up"
    data-testid="ASC"
  />
  <span
    className="sort__icon-down"
    data-testid="DESC"
  />
</span>
`;
